src/main/java/controllers/BookstoreRestController.java

```diff
@@ -0,0 +1,177 @@
+package controllers;
+
+import models.*;
+import repositories.*;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+public class BookstoreRestController {
+
+    @Autowired
+    private BookRepository bookRepository;
+    @Autowired
+    private BookstoreRepository bookstoreRepository;
+    @Autowired
+    private BookstoreOwnerRepository bookstoreOwnerRepository;
+    @Autowired
+    private CustomerRepository customerRepository;
+    @Autowired
+    private ShoppingCartRepository shoppingCartRepository;
+    @Autowired
+    private SaleRepository saleRepository;
+
+
+    //BookstoreOwner REST endpoints
+    @GetMapping("/api/getBookstoreOwner")
+    public BookstoreOwner getBookstoreOwner(@RequestParam(value = "bookstoreOwnerId") long bookstoreOwnerId) {
+        return bookstoreOwnerRepository.findById(bookstoreOwnerId);
+    }
+
+    @GetMapping("/api/getBookstoreOwners")
+    public Iterable<BookstoreOwner> getBookstoreOwner() {
+        return bookstoreOwnerRepository.findAll();
+    }
+
+    @PostMapping("/api/newBookstoreOwner")
+    public BookstoreOwner newBookstoreOwner(@RequestParam(value = "bookstoreOwnerName") String bookstoreOwnerName) {
+        BookstoreOwner bookstoreOwner = new BookstoreOwner(bookstoreOwnerName);
+        bookstoreOwnerRepository.save(bookstoreOwner);
+        return bookstoreOwner;
+    }
+
+
+    //Bookstore REST endpoints
+    @GetMapping("/api/getBookstore")
+    public Bookstore getBookstore(@RequestParam(value = "bookstoreId") long bookstoreId) {
```

Comment:
As per our discussion, let's change this to getBookstoreByBookstoreOwner(). This is to create a standardized REST API and to avoid circular references due to inability to selectively use @JsonIgnore when querying for entity JSON

Reply:
Addressed in dbd44c35c5fca73c33c85056fd10ce02cac08a53 and 5218b27cecbc7f1eca581a07f2781581b341df49